cloudpickle/cloudpickle.py

```diff
@@ -1090,12 +1107,17 @@ def _make_skel_func(code, cell_count, base_globals=None):
     if base_globals is None:
         base_globals = {}
     elif isinstance(base_globals, str):
+        base_globals_name = base_globals
         if sys.modules.get(base_globals, None) is not None:
-            # this checks if we can import the previous environment the object
+            # This checks if we can import the previous environment the object
             # lived in
             base_globals = vars(sys.modules[base_globals])
         else:
-            base_globals = {}
+            base_globals = _dynamic_modules_globals.get(
+                    base_globals, DynamicDict())
+            # base_globals is not a string anymore, using base_globals_name
+            # instead
```

Comment:
This comment is unnecessary in my opinion. The `base_globals_name` variable name is explicit enough.